onnx/common/ir.h

```diff
@@ -52,7 +52,10 @@ class ResourceGuard final {
   bool released_;
 
  public:
-  ResourceGuard(std::function<void()> destructor) : destructor_(std::move(destructor)), released_(false) {}
+  ONNX_DISALLOW_COPY_AND_ASSIGN(ResourceGuard);
```

Comment:
Bug: ResourceGuard should not be copy-able, for the `destructor` function can only be executed once.